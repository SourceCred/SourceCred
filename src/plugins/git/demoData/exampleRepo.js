// @flow

import mkdirp from "mkdirp";

import {makeUtils} from "../gitUtils";
import type {Hash} from "../types";

type RepositoryInfo = {|
  +path: string,
  +commits: $ReadOnlyArray<Hash>, // in oldest-to-newest order
|};

export function createExampleRepo(intoDirectory: string): RepositoryInfo {
  const repositoryPath = intoDirectory;
  mkdirp(repositoryPath);
  const git = makeUtils(repositoryPath);
  const commits = [];

  git.exec(["init"]);

  git.writeAndStage("README.txt", "Amazing physics going on...\n");
  git.deterministicCommit("Initial commit");
  commits.push(git.head());

  git.writeAndStage("src/index.py", "import antigravity\n");
  git.writeAndStage(
    "src/quantum_gravity.py",
    'raise NotImplementedError("TODO(physicists)")\n'
  );
  git.writeAndStage("TODOS.txt", "1. Resolve quantum gravity\n");
  git.deterministicCommit("Discover gravity");
  commits.push(git.head());

  git.writeAndStage(
    "src/quantum_gravity.py",
    "import random\nif random.random() < 0.5:\n  import antigravity\n"
  );
  git.deterministicCommit("Solve quantum gravity");
  commits.push(git.head());

  git.exec(["rm", "TODOS.txt"]);
  git.deterministicCommit("Clean up TODOS");
  commits.push(git.head());

  return {path: repositoryPath, commits};
}
