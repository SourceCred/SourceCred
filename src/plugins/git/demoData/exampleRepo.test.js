// @flow

import tmp from "tmp";

import {createExampleRepo} from "./exampleRepo";

const cleanups: (() => void)[] = [];
afterAll(() => {
  cleanups.forEach((f) => {
    f();
  });
});

function mkdtemp() {
  const result = tmp.dirSync({unsafeCleanup: true});
  cleanups.push(() => result.removeCallback());
  return result.name;
}

describe("createExampleRepo", () => {
  it("is deterministic", () => {
    expect(createExampleRepo(mkdtemp()).commits).toMatchSnapshot();
  });
});
