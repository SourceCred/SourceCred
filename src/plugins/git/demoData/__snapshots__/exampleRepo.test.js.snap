// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createExampleRepo is deterministic 1`] = `
Array [
  "301749e9af8cd6e9aee3a49a64029b98a4695e34",
  "cbb26b570d1eed3c681b8f03ff31231c1bffd6d6",
  "4be43f1cda04e51e42fec0cfe8e1e2dff116e839",
  "677b340674bde17fdaac3b5f5eef929139ef2a52",
]
`;
